### PR TITLE
BaSP-MR-89: Subscription controller added populate

### PR DESCRIPTION
**Pull request by jramire5**
Added populate for get-By-Id and simplified tests to make them work with the new changes

### CODE REVIEW

File: src/controllers/subscription.js [hunk 1]
```diff
@@ -34,25 +33,23 @@ const getAllSubs = (req, res) => {
...
-        error: false,
-      });
-    })
-    .catch((error) => {
-      return res.status(500).json({
-        message: 'Internal error',
-        error,
-      });
-    });
+    .then((subscriptions) => res.status(200).json({
```
Comment (IgnacioGalceran) [inline src/controllers/subscription.js:36]: this needs a return here

File: src/controllers/subscription.js [hunk 2]
```diff
@@ -34,25 +33,23 @@ const getAllSubs = (req, res) => {
...
+      message: 'All subscriptions list',
+      data: subscriptions,
+      error: false,
+    }))
+    .catch((error) => res.status(500).json({
```
Comment (IgnacioGalceran) [inline src/controllers/subscription.js:41]: and here

File: src/tests/subscriptions.test.js [hunk 3]
```diff
@@ -57,9 +57,9 @@ describe('GET /api/subscription/:id', () => {
       .get(`/api/subscription/${mockSubId}`);
     expect(response.status).toBe(200);
     expect(response.error).toBeFalsy();
-    expect(response.body.data.classes).toBe(subsSeed[0].classes);
-    expect(response.body.data.members).toBe(subsSeed[0].members);
-    expect(response.body.data.date).toBe(subsSeed[0].date);
```
Comment (IgnacioGalceran) [inline src/tests/subscriptions.test.js:62]: this tests didn't passed? why?

Comment (jramire5) [reply src/tests/subscriptions.test.js:62]: because they didn't check the populated classes and members, they were just checking the ID

**Review by santimenendez98 — APPROVED** [review]

**Pull request merged**